src/Microsoft.AspNet.Authentication.OAuth/OAuthAuthenticationMiddleware.cs

```diff
@@ -43,27 +43,27 @@ public class OAuthAuthenticationMiddleware<TOptions, TNotifications> : Authentic
             // todo: review error handling
             if (string.IsNullOrWhiteSpace(Options.AuthenticationScheme))
             {
-                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options.AuthenticationScheme)));
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options)));
```

Comment:
FYI the ```nameof``` issue has been suggested here https://github.com/dotnet/roslyn/issues/3759